src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/Navigation.java

```diff
@@ -319,31 +279,40 @@ private void addMainNavLinks(Content target) {
             case EXTERNAL_SPECS:
             case RESTRICTED:
             case SERIALIZED_FORM:
-            case SEARCH:
             case SYSTEM_PROPERTIES:
                 addOverviewLink(target);
-                addModuleLink(target);
-                addPackageLink(target);
-                addPageLabel(target, contents.classLabel, true);
-                addPageLabel(target, contents.useLabel, options.classUse());
                 addTreeLink(target);
                 addPreviewLink(target);
                 addNewLink(target);
                 addDeprecatedLink(target);
                 addIndexLink(target);
+                addSearchLink(target);
                 addHelpLink(target);
                 break;
             case DOC_FILE:
                 addOverviewLink(target);
-                addModuleOfElementLink(target);
-                addItemToList(target, navLinkPackage);
-                addPageLabel(target, contents.classLabel, true);
-                addPageLabel(target, contents.useLabel, options.classUse());
-                addTreeLink(target);
+                if (element != null) {
+                    addPageElementLink(target);
+                    if (options.classUse()) {
+                        if (element instanceof PackageElement pkg) {
```

Comment:
Yeah ... this is where it is nice to use `instanceof` patterns, even though the recommended form is to check the `ElementKind` and do a separate cast.  As I said above, this will work on JDK, but the coding style may not work in other contexts.